checkout 10_9 paddle commit && complie with 2023 api

fc657547899 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->

### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->

### Describe
<!-- Describe what this PR does -->

fc657547899 committed:
checkout 10_9 paddle commit && complie with 2023 api

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=49545) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=49545) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=49545) it.</sub>

gaodaheng committed:
Add custom_fused_dense for mlperf bert train.

liyupeng committed:
Add custom_fmha for mlperf bert train, NO XPUAPI implement, need fix

gaodaheng committed:
Add AccMergeOp and CustomLrOp of bert train.

fc657547899 committed:
custom_fused_dropout_residual_ln float support

fc657547899 committed:
fix custom_fused_dropout_residual_ln_op grad op init error

liyupeng committed:
fix bugs and refine

fc657547899 committed:
fix lookuptable_v2, gather, custom_lr error

gaodaheng committed:
Add acc_merge xpu op.

gaodaheng committed:
Merge branch 'api_mlperf_bert' of github.com:/fc657547899/Paddle into api_mlperf_bert

fc657547899 committed:
temp pass custom_fmha custom_fused_dropout_residual_ln cross_entropy scatter

fc657547899 committed:
bind scatter_grad; skip equal, custom_fused_dropout_residual_ln grad;
tmp resolve cross_entropy_grad

fc657547899 committed:
fix reduce_sum, cross_entropy, embedding grad ...

fc657547899 committed:
fix complie error

fc657547899 committed:
bind custom_fused_dropout_residual_ln

liyupeng committed:
bind fmha kernel

liyupeng committed:
fix python interface bug of bind fmha kernel

liyupeng committed:
fix syntax bug of fmha python interface

fc657547899 committed:
cast fmha host_seq_len to cpu

fc657547899 committed:
cast fmha host_seq_len to cpu

fc657547899 committed:
fix custom_fmha_grad tensor DQKV name error

fc657547899 committed:
custom_fmha_grad error xpu_memcpy

fc657547899 committed:
support print op time,
check fmha,
update api compile error,
support arg_max, idx_select fp16
fix equal op error
add scatter, scatter_grad cal

fc657547899 committed:
To do:index_select_grad support fp16 big data

gaodaheng committed:
Add fc bias fusion.

fc657547899 committed:
limit fp16, modify loss_scaling min value  1/256to avoid inf and nan

Commit:
Add dbias fusion optimization.

fc657547899 committed:
add distributed_fused_lamb, Todo: check, custom_lr

Commit:
Fix fc dbias fusion dbiasType not support.

fc657547899 committed:
temp solve inf&&nan distributed_fused_lamb null ptr

Commit:
Cancle fc dbias fusion.

Commit:
Fmha grad L3 optimization.

fc657547899 committed:
fix && optimize xpu distributed_fused_lamb
support acc_step > 1

Zhangjingyu06 committed:
add dqkv param and remove concat in fmha_grad

fc657547899 committed:
distributed_fused_lamb support elementwiseadd_with_cast xdnn op

fc657547899 committed:
support dropout_residual_ln mask bit16 type, modify all weight alloc alignment

Zhangjingyu06 committed:
add l3 use in custom_fuse_dense

fc657547899 committed:
xpu support index_select_grad

gaodaheng committed:
Move fused_dense to api and optimize.
